web/packages/core/src/load-options.ts

```diff
@@ -319,6 +352,20 @@ export interface BaseLoadOptions {
      */
     playerVersion?: number | null;
 
+    /**
+     * The preferred render backend of the Ruffle player.
```

Comment:
I'd add to this that a preference does not guarantee that specific backend, only request it if it's available.

Though, if we're using this as a testing feature, maybe changing that would actually be smart and having it be "force" instead of "prefer"; I know I've been caught a few times by using one backend instead of another when the preference wasn't doing what I expected during dev. This backend or error, so you know it's what you chose?

Reply:
How easy is it to detect and determine at runtime which backend is actually used? I wonder if the current functionality could be kept, but then have both the "preferred" and the "actually used" backends being shown somewhere that is easy to access (I do not know where that "somewhere" would be, though).

![beautiful_mspaint_illustration](https://user-images.githubusercontent.com/131509408/233870360-cd71e628-ce13-406a-aaee-5ee51ba873fe.png)

Like in this image, where the two left-side labels do not change, and the two right-side labels change depending on which backend is preferred and which backend is currently used respectively.

I guess that this showing of "preferred" and "currently used" could come in a later PR, though.

Reply:
I wonder whether a checkbox option could be added (either now or in a later PR), where you 'force' the backend used or 'give error' if the preferred backend is different from the currently used backend. Such that, in the regular behavior as it is implemented now, it is just preferred, but when that option is chosen, an error is given if the preferred backend is not the same as the currently used backend.

Reply:
> having it be "force" instead of "prefer"

Too late for that suggestion, sorry

> I guess that this showing of "preferred" and "currently used" could come in a later PR, though.

Yep and that PR won't be done by me 😛 

> I wonder whether a checkbox option could be added (either now or in a later PR), where you 'force' the backend used

Separate PR that will not be done by me

Reply:
> Yep and that PR won't be done by me :stuck_out_tongue:

> Separate PR that will not be done by me

That is alright, sorry if it sounded like I implied that it should be you, I did not mean it like that. Piling substantially more work on you as a contributor would not be fair.

I am more interested if you think this proposed functionality might be doable (for instance by someone else). Because, if this proposed functionality can be done in a later PR, then that other functionality can always be decided upon later, for instance after this PR has been merged, and thus this discussion would not block merging this PR. I meant to be helpful, sorry if I came off as something else.

Reply:
No worries, I'm just a bit irritable about the whole process I've gone through to get this PR accepted lol.

Yes, a "force renderer or panic" config option would definitely be doable. Would just be an extra boolean that is checked in
web/src/lib.rs (fn create_renderer) and changes its behavior.

Reply:
@n0samu Wait, why not acutally implement "force renderer" in this PR? I agree with Dinnerbone that "preferred renderer" is not ideal for testing purposes.

Reply:
> Wait, why not acutally implement "force renderer" in this PR?

Because it's a separate functionality, with - in a way - completely separate philosophy from the original intent.

Can we please not restart this discussion this late in? It was basically hours from being merged after being stuck in limbo for over month.

Reply:
Main thing is it would be a lot of extra work and I'm not going to do it lol